packages/camera/camera_platform_interface/lib/src/platform_interface/camera_platform.dart

```diff
@@ -0,0 +1,115 @@
+  static set instance(CameraPlatform instance) {
+    PlatformInterface.verifyToken(instance, _token);
+    _instance = instance;
+  }
+
+  /// Completes with a list of available cameras.
+  Future<List<CameraDescription>> availableCameras() {
+    throw UnimplementedError('availableCameras() is not implemented.');
+  }
+
+  /// Initializes the camera on the device and returns its textureId.
+  Future<int> initializeCamera(
+    CameraDescription cameraDescription, {
+    ResolutionPreset resolutionPreset,
+    bool enableAudio,
+  }) {
+    throw UnimplementedError('initializeCamera() is not implemented.');
+  }
+
+  /// Returns a Stream of [CameraEvent]s.
+  Stream<CameraEvent> cameraEventsFor(int textureId) {
+    throw UnimplementedError('videoEventsFor() has not been implemented.');
+  }
+
+  /// Captures an image and saves it to [path].
+  Future<void> takePicture(int textureId, String path) {
+    throw UnimplementedError('takePicture() is not implemented.');
+  }
+
+  /// Prepare the capture session for video recording.
+  Future<void> prepareForVideoRecording() {
+    throw UnimplementedError('prepareForVideoRecording() is not implemented.');
+  }
+
+  /// Start a video recording and save the file to [path].
+  ///
+  /// A path can for example be obtained using
+  /// [path_provider](https://pub.dartlang.org/packages/path_provider).
+  ///
+  /// The file is written on the flight as the video is being recorded.
+  /// If a file already exists at the provided path an error will be thrown.
+  /// The file can be read as soon as [stopVideoRecording] returns.
+  Future<void> startVideoRecording(int textureId, String path) {
+    throw UnimplementedError('startVideoRecording() is not implemented.');
+  }
+
+  /// Stop the video recording.
+  Future<void> stopVideoRecording(int textureId) {
+    throw UnimplementedError('stopVideoRecording() is not implemented.');
+  }
+
+  /// Pause video recording.
+  Future<void> pauseVideoRecording(int textureId) {
+    throw UnimplementedError('pauseVideoRecording() is not implemented.');
+  }
+
+  /// Resume video recording after pausing.
+  Future<void> resumeVideoRecording(int textureId) {
+    throw UnimplementedError('resumeVideoRecording() is not implemented.');
+  }
+
+  /// Start streaming images from platform camera.
+  ///
+  /// When running continuously with [CameraPreview] widget, this function runs
+  /// best with [ResolutionPreset.low]. Running on [ResolutionPreset.high] can
+  /// have significant frame rate drops for [CameraPreview] on lower end
+  /// devices.
+  // TODO(bmparr): Add settings for resolution and fps.
+  Future<void> startImageStream(onLatestImageAvailable onAvailable) {
+    throw UnimplementedError('startImageStream() is not implemented.');
+  }
+
+  /// Stop streaming images from platform camera.
+  Future<void> stopImageStream() {
+    throw UnimplementedError('stopImageStream() is not implemented.');
+  }
```

Comment:
I think converting this from a method with a callback to a `Stream<CameraImage>` makes the api easier to consume.

I am not sure why there is no `textureId` at the moment. @bparrishMines do you know if there is a special reason or was it simply overlooked and would it make sense to add one while we are refactoring the camera plugin?

Reply:
(Was this the one discussed offline, and determined that it might disappear from the interface altogether? Or not be web implementable?)

Reply:
Yes sorry, Amir mentioned he would inform you. 

We decided that the `startImageStream` and `stopImageStream` methods need to be implemented in a different way al together. The current implementation in not very efficient since it will stream all data first to Dart and then in most cases back to the native side on a different plugin (e.g. QR scanner plugin). The idea is to redesign the feature to keep everything on the native side.

However since at the moment we only focus on the basic camera features (taking a picture and recording video), we decided to not include the above methods in the platform interface. To not break the Android and iOS implementations we will keep the methods there.